core/dbt/parser/manifest.py

```diff
@@ -478,6 +478,12 @@ def write_manifest_for_partial_parse(self):
         path = os.path.join(self.root_project.target_path,
                             PARTIAL_PARSE_FILE_NAME)
         try:
+            # This shouldn't be necessary, but we have gotten bug reports (#3757) of the
+            # saved manifest not matching the code version.
+            if self.manifest.metadata.dbt_version != __version__:
+                logger.debug("Manifest metadata did not contain correct version. "
+                             "Contained '{self.manifest.metadata.dbt_version}' instead.")
+                self.manifest.dbt_version = __version__
```

Comment:
Should this be
```suggestion
                self.manifest.metadata.dbt_version = __version__
```

Reply:
Arggh... Hard to test, but should have tried harder :)